setup.py

```diff
@@ -69,6 +69,7 @@ def read_requirements(filename):
     "docker",
     "tqdm",
     "psutil",
+    "ipython~=8.18.1",
```

Comment:
Are they ok with this? This is adding dependency to core pysdk right?

Reply:
we should remove this dependency and use print method for logging. the IPython is used for notebook environment only.

Reply:
+1 please remove these dependencies, also they need to be pinned to versions, and i dont think they will allow this.